### PR TITLE
569 multithreaded assigner

### PR DESCRIPTION
**Pull request by CblPOK-git**

### CODE REVIEW

File: bin/assigner/src/main.cpp [hunk 1]
```diff
@@ -566,23 +636,21 @@ int curve_dependent_main(std::string bytecode_file_name,
               (target_prover < assigner_instance.assignments.size() || target_prover == invalid_target_prover)) {
         std::uint32_t start_idx = (target_prover == invalid_target_prover) ? 0 : target_prover;
         std::uint32_t end_idx = (target_prover == invalid_target_prover) ? assigner_instance.assignments.size() : target_prover + 1;
+
+        std::vector<std::thread> threads = {};
```
Comment (akokoshn) [inline bin/assigner/src/main.cpp:640]: May be make sense to use `std::future` instead `threads` (https://en.cppreference.com/w/cpp/thread/future). Automatically join when go out of the local scope

File: bin/assigner/src/main.cpp [hunk 2]
```diff
@@ -156,11 +159,62 @@ enum class print_column_kind {
     SELECTOR
 };
 
-template<typename ValueType, typename ContainerType>
-void fill_vector_value(std::vector<ValueType> &table_values, const ContainerType &table_col, typename std::vector<ValueType>::iterator start) {
-    std::copy(table_col.begin(), table_col.end(), start);
+template<typename Endianness>
+void print_size_t(
```
Comment (akokoshn) [inline bin/assigner/src/main.cpp:163]: why we can't use `crypto3::marshaling` write implementation: https://github.com/NilFoundation/marshalling/blob/c5a00412ba13be6097b921884b424897887dd286/include/nil/marshalling/types/integral/basic_type.hpp#L110

Comment (CblPOK-git) [reply bin/assigner/src/main.cpp:177]: I tried, seems like it doesn't work with ofstreams, only with arrays

Comment (CblPOK-git) [reply bin/assigner/src/main.cpp:177]:
```
error: no type named 'iterator_category' in 'std::iterator_traits<std::basic_ostream<char>>'
[build]                             = std::is_same<typename std::iterator_traits<TIter>::iterator_category,
[build]                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```

Comment (akokoshn) [reply bin/assigner/src/main.cpp:177]: expected, it doesn't works as is. Need to add implementation read/write for `ofstream::iterator`

**Review by akokoshn — APPROVED** [review]

**Pull request merged**